oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/Types.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.oneandone.rest.domain;
+
+import com.google.common.base.Enums;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Types {
+
+    public enum OSFamliyType {
+
+        Windows("Windows"), Linux("Linux"), Others("Others"), Null("Null"), UNRECOGNIZED("");
+
+        public static OSFamliyType fromValue(String v) {
+            return Enums.getIfPresent(OSFamliyType.class, v).or(UNRECOGNIZED);
+        }
+
+        // the value which is used for matching
+        // the json node value with this enum
+        private final String value;
+
+        OSFamliyType(final String type) {
+            value = type;
+        }
+
+        @Override
+        public String toString() {
+            return value;
```

Comment:
Do we need to actually "produce" the empty string as the enum value, when it is UNRECOGNIZED? If not, you can get rid of the value variable, the constructor and this method, as the default enum implementation will use the `name()` method.